synchronizer/synchronizer.go

```diff
@@ -315,6 +333,31 @@ func (s *Synchronizer) AdvanceView(syncInfo hotstuff.SyncInfo) {
 		}
 	}
 
+	b, ok := syncInfo.Block()
```

Comment:
Perhaps this could be made into a separate method to reduce the method's lines of code and its cyclomatic complexity.

Reply:
moved to updateLeafBlock

Reply:
Sorry, I wasn't clear enough in my message. I suggested moving all of the new code related to pipelining into a separate method, not just this, but it did help I guess.